app/views/unattended/provisioning_templates/iPXE/kickstart_default_ipxe.erb

```diff
@@ -40,7 +40,7 @@ ping --count 1 ${netX/gateway} || echo Ping to Gateway failed or ping command no
 echo Trying to ping DNS: ${netX/dns}
 ping --count 1 ${netX/dns} || echo Ping to DNS failed or ping command not available.
 
-kernel <%= "#{@host.url_for_boot(:kernel)}" %> initrd=initrd.img ks=<%= ks %> <%= pxe_kernel_options %> <%= snippet("kickstart_kernel_options", variables: {ipxe_net: true}).strip %>
```

Comment:
If we remove this entry, do we still need https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/iPXE/kickstart_default_ipxe.erb#L25-L35 block?

Also, I'm not a provisioning guy, but what's the difference between https://github.com/theforeman/foreman/blob/0776ee953497a9772a4ca1bd86c68d9c5cf9051a/app/views/unattended/provisioning_templates/iPXE/kickstart_default_ipxe.erb#L30 and https://github.com/theforeman/foreman/blob/0776ee953497a9772a4ca1bd86c68d9c5cf9051a/app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb#L39 ?

Reply:
> If we remove this entry, do we still need https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/iPXE/kickstart_default_ipxe.erb#L25-L35 block?
> 
> Also, I'm not a provisioning guy, but what's the difference between
> 
> https://github.com/theforeman/foreman/blob/0776ee953497a9772a4ca1bd86c68d9c5cf9051a/app/views/unattended/provisioning_templates/iPXE/kickstart_default_ipxe.erb#L30
> and
> 
> https://github.com/theforeman/foreman/blob/0776ee953497a9772a4ca1bd86c68d9c5cf9051a/app/views/unattended/provisioning_templates/snippet/kickstart_kernel_options.erb#L39
> ?

I'll preface this with admitting to be a fairly infrequent contributor, and not an expert in Ruby templates. This may go more under the hood with foreman_url() and various kickstart deployment scenarios than I am able to speak on.  :) 

RE: kickstart_default_ipxe.erb#L25-L35
That being said, I believe you're right.  kickstart_default_ipxe.erb Line 29 - 36 looks to be setting the value of ks and is making determinations based on if it has a static IPv4/IPv6 address set or is DHCP.  As far as I know, foreman_url('provisioning', static: '1') would render the foreman url with the static=1 url parameter.  I haven't seen this on ks= urls in my deployment, but I do not have Satellite integrating with DNS & DHCP services where maybe this comes into play?

I think the kickstart_default_ipxe.erb L25-35 block could be integrated with the kickstart_kernel_options L36-40 If/Else block, if needed, and should arrive at a workable result.

Reply:
Hmm, I agree with your proposal since it would allow us to be safer with the changes. Thus said, let's move `ks` variable setting from `kickstart_default_ipxe` into `kickstart_kernel_options` snippet. This way we don't actually remove the logic and we wouldn't set `ks` variable up more than once (as I see this PR is about). Could you please update?

Reply:
Will do!  I couldn't get this today, but will try to get it in tomorrow.